<launch>
  <!-- INPUT ARGUMENTS -->
  <!-- Mode for training spatial models this session: May be set to "new", "aggregate" or "off" -->
  <arg name="train_mode" default="new"/>
  <!-- Whether to save newest train data to corpus_file -->
  <arg name="save_train_data" default="false"/>
  <!-- Location in which to save/retrieve processed data -->
  <arg name="corpus_file" default="$(find hrc_discrim_learning)/train/feature_corpus.csv"/>\
  <!-- Location of trained spatial model -->
  <arg name="spatial_model" default="$(find hrc_discrim_learning)/model/spatial"/>

  <!-- Whether to use the perception stuff -->
  <arg name="use_perception" default="false"/>
  <!-- File containing object locations, if not using perception -->
  <arg name="env_file" default="$(find hrc_discrim_learning)/train/full_envs.json"/>
  <!-- File containing newest training (id, utterance) data -->
  <arg name="train_data_file" default="$(find hrc_discrim_learning)/train/spoken_descriptions.json"/>
  <!-- Whether to save the model trained this session -->
  <arg name="save_model" default="false"/>
  <!-- If save_model = True, folder in which to save model -->
  <arg name="save_dest" default="$(find hrc_discrim_learning)/model/features"/>

  <!-- ROS PARAMETERS -->
  <param name="hrc_discrim_learning/use_perception"     value="$(arg use_perception)"/>
  <param name="hrc_discrim_learning/env_file"           value="$(arg env_file)"/>

  <param name="hrc_discrim_learning/spatial_model"      value="$(arg spatial_model)"/>

  <param name="hrc_discrim_learning/save_model"         value="$(arg save_model)"/>
  <param name="hrc_discrim_learning/save_dest"          value="$(arg save_dest)"/>
  <param name="hrc_discrim_learning/train_mode"         value="$(arg train_mode)"/>
  <param name="hrc_discrim_learning/train_data_file"    value="$(arg train_data_file)"/>
  <param name="hrc_discrim_learning/save_train_data"    value="$(arg save_train_data)"/>
  <param name="hrc_discrim_learning/corpus_file"        value="$(arg corpus_file)"/>



  <!-- LAUNCH -->
  <group if="$(arg use_perception)">
    <include file="$(find hrc_discrim_learning)/launch/basic_perception.launch"/>
  </group>

  <node pkg="hrc_discrim_learning" type="auto_train.py" name="scripted_training" respawn="false" output="screen"/>

  <node pkg="hrc_discrim_learning" type="trainer_features.py" name="train_features" respawn="false" output="screen"/>

</launch>
